src/main/java/io/github/jhipster/online/service/GithubService.java

```diff
@@ -262,12 +221,70 @@ private GitHub getConnection(User user) throws Exception {
     }
 
     /**
-     *  Delete all the github organizations.
+     * Delete all the github organizations.
      *
      */
     @Transactional
     public void deleteAllOrganizationsUser(User user) {
         log.debug("Request to delete all github organizations for user {}", user.getLogin());
         gitCompanyRepository.deleteAllByUserLoginAndGitProvider(user.getLogin(), GitProvider.GITHUB.getValue());
     }
+
+    /**
+     *  Sync the projects from the user's companies
+     */
+    private void syncUserProjectsFromCompanies(User user, GitHub gitHub, Set<GitCompany> currentGithubCompanies, Set<GitCompany> updatedGithubCompanies) throws IOException {
+        Map<String, GHOrganization> githubOrganizations = gitHub.getMyOrganizations();
+        for (String githubOrganizationName : githubOrganizations.keySet()) {
+            log.debug("Syncing company `{}`", githubOrganizationName);
```

Comment:
@gmarziou : Thanks for the review. I have corrected everything you mentioned except for this one. Could you provide some more details on how to do this? I haven't used MDC before. :thinking:

Reply:
Here is an article that gives an example: https://dzone.com/articles/mdc-better-way-of-logging-1

Basically, MDC is a Map where you can store values and that logback can access by names.There are some pitfalls though, MDC is bounded to the current thread. So, in case of  a web API, you could use a servlet filter or a handler interceptor to put value and clean it up at the end of the request processing but if this processing involves another thread like with an async executor service the MDC won't be propagated to the aysnc thread unless you do it explicitly.

Reply:
@gmarziou : I've added the explicit MDC variable in the `syncUserProjectsFromCompanies`. Take a look and let me know if everything looks good or needs adjustment. Thanks. :smile: